### PR TITLE
Remove uncurried chapter and update existing docs to uncurried

### PR DESCRIPTION
**Pull request by fhammerschmidt**
Also removed a lot of final units in example code.

### CODE REVIEW

File: misc_docs/syntax/decorator_uncurried.mdx [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+If you have uncurried mode turned off in `rescript.json` and still want to try it on a per-file basis, you can turn it on via 
+
+```rescript
+@@uncurried
+```
+
+at the top of a `.res` file.
+
+_Available since ReScript `11.0.0`._
```
Comment (ryyppy) [inline misc_docs/syntax/decorator_uncurried.mdx:17]: Any reference link to our blog post / guide as we do in our other cheatsheets would be useful

**Review by zth — APPROVED** [review]

File: pages/docs/manual/latest/build-configuration.mdx [hunk 2]
```diff
@@ -7,7 +7,7 @@ canonical: "/docs/manual/latest/build-configuration"
 
 # Configuration
 
-`rescript.json` (or `rescript.json` in versions prior ReScript 11) is the single, mandatory build meta file needed for `rescript`.
+`rescript.json` (or `bsconfig.json` in versions prior ReScript 11) is the single, mandatory build meta file needed for `rescript`.
```
[the remark this answers is not in the document]
Comment (fhammerschmidt) [reply pages/docs/manual/latest/build-configuration.mdx:10]: Yes? that's what it was called before.

[the remark this answers is not in the document]
Comment (ryyppy) [reply pages/docs/manual/latest/build-configuration.mdx:10]: I saw that too when the previous PR was merged, the proposed change is correct.

**Review by ryyppy — CHANGES_REQUESTED** [review]
A few comments / nitpicks. Bit generally, lookibg great!

File: pages/docs/manual/latest/build-configuration.mdx [hunk 3]
```diff
@@ -152,7 +139,19 @@ This configuration only applies to you, when you develop the project. When the p
...
+- `".mjs"`
+- `".cjs"`
+- `".res.js"`
+- `".res.mjs"`
+- `".res.cjs"`
+- `".bs.js"`
+- `".bs.mjs"`
+- `".bs.cjs"`
+
+Currently prefer `.bs.js` for now.
```
[the remark this answers is not in the document]
Comment (ryyppy) [reply pages/docs/manual/latest/build-configuration.mdx:154]: I think the main concern was mostly about genType relying on bs.js when I am not mistaken? Nowadays genType reads the rescript.json anyways, so I assume some constraints regarding configuring the .js extension should be lifted?

**Review by ryyppy — APPROVED** [review]
Looks good! Only question I have is how we treat the (.) => {} syntax in the docs.

As I understand, everything is now an uncurried function. But at some point ppl will see the syntax and wonder what it is.

**Comment by fhammerschmidt** [comment]
Suggestion: 
- Remove all occurences where `(.)` it is used (e.g. this-based callbacks)
- Mention in https://rescript-lang.org/syntax-lookup#uncurried-function that it is only relevant for <11 or if uncurried is deactivated explicitly and also mention to use the older docs if one prefers currying. Not ideal, but it should make it clear that uncurried is the default way in ReScript going forward.
- Also https://rescript-lang.org/syntax-lookup#uncurry-decorator is pretty much useless now, that chapter can also have such a warning

**Pull request merged**